opts.go

```
@@ -47,3 +48,10 @@ func WithProcessRunner(cmd *exec.Cmd) Opt {
 		machine.cmd = cmd
 	}
 }
+
+// WithSnapshot will allow for the machine to start using a given snapshot.
+func WithSnapshot(ctx context.Context, memFilePath, snapshotPath string) Opt {
+	return func(machine *Machine) {
+		machine.LoadSnapshot(ctx, memFilePath, snapshotPath)
```

We must not ignore the error here.